extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/SaslConfig.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.kafka.streams.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class SaslConfig {
+
+    /**
+     * JAAS login context parameters for SASL connections in the format used by JAAS configuration files
+     */
+    @ConfigItem(name = "jaas.config")
+    public Optional<String> jaasConfig;
+
+    /**
+     * The fully qualified name of a SASL client callback handler class
+     */
+    @ConfigItem(name = "client.callback.handler.class")
+    public Optional<String> clientCallbackHandlerClass;
+
+    /**
+     * The fully qualified name of a SASL login callback handler class
+     */
+    @ConfigItem(name = "login.callback.handler.class")
+    public Optional<String> loginCallbackHandlerClass;
+
+    /**
+     * The fully qualified name of a class that implements the Login interface
+     */
+    @ConfigItem(name = "login.class")
+    public Optional<String> loginClass;
+
+    /**
+     * The Kerberos principal name that Kafka runs as
+     */
+    @ConfigItem(name = "kerberos.service.name")
+    public Optional<String> kerberosServiceName;
+
+    /**
+     * Kerberos kinit command path
+     */
+    @ConfigItem(name = "kerberos.kinit.cmd")
+    public Optional<String> kerberosKinitCmd;
+
+    /**
+     * Login thread will sleep until the specified window factor of time from last refresh
+     */
+    @ConfigItem(name = "kerberos.ticket.renew.window.factor")
+    public Optional<String> kerberosTicketRenewWindowFactor;
```

Comment:
Why is this a string, shouldn't it be rather a double (some for some more props below)?

Reply:
Yeah, I didn't bother, since KS's config takes care of the conversions anyway.
If you think we should, then this can easily be changed.

Reply:
Let's ask @gsmet. My understanding is that Quarkus wants to expose properly typed options, so to offer e.g. editor support (validation) down the road.

Reply:
Yes, the Quarkus config should be properly typed.

Reply:
Yes, it should be typed.